core/blockchain.go

```diff
@@ -792,15 +792,14 @@ func (bc *BlockChain) WriteBlockAndState(block *types.Block, receipts []*types.R
 	if ptd == nil {
 		return NonStatTy, consensus.ErrUnknownAncestor
 	}
+	// If the block is already known, skip it
+	if bc.HasBlockAndState(block.Hash()) {
+		return SideStatTy, nil
```

Comment:
The previous return value for this case was `NonStatTy`.

Reply:
I'm starting to winder whether this is meaningful at all. We have https://github.com/ethereum/go-ethereum/blob/master/core/blockchain.go#L931 here to dedup already, so this entire check in Write... seems just moot.

Reply:
Does it work if you just remove the check?